sgt-back-end/index.js

```diff
@@ -0,0 +1,178 @@
+const express = require('express');
+const app = express();
+
+app.listen(3000, () => {
+  // eslint-disable-next-line no-console
+  console.log('Express server listening on port 3000');
+});
+
+const pg = require('pg');
+
+// only create ONE pool for your whole server
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.get('/api/grades', (req, res) => {
+  const sql = `
+  select *
+    from "grades"
+  `;
+  db.query(sql)
+    .then(result => {
+      const grades = result.rows;
+      res.status(200).json(grades);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({ error: 'An unexpected error occurred' });
+    });
+});
+
+app.post('/api/grades', (req, res) => {
+
+  const name = req.body.name;
+  const course = req.body.course;
+  const score = Number(req.body.score);
+  const params = [name, course, score];
+
+  if (!name && !course) {
+    res.status(400).json({
+      error: 'name & course are required'
+    });
+    return;
+  } else if (!course) {
+    res.status(400).json({
+      error: 'course is required'
+    });
+    return;
+  } else if (!name) {
+    res.status(400).json({
+      error: 'name is required'
+    });
+    return;
+  } else if (!Number.isInteger(score) || score < 0) {
+    res.status(400).json({
+      error: 'score must be a positive integer'
+    });
+    return;
+  }
```

Comment:
A `score` over `100` should also result in a `400 Bad Request` response:

---

![image](https://user-images.githubusercontent.com/16143771/162357599-7b8fc3c0-2bc3-4d21-aaec-7e77b9a57872.png)

Reply:
@TimelikeClosure updated the range to be 0 to 100